pkg/spiffe/spiffe.go

```diff
@@ -272,15 +272,18 @@ func (v *PeerCertVerifier) AddMappings(certMap map[string][]*x509.Certificate) {
 // VerifyPeerCert is an implementation of tls.Config.VerifyPeerCertificate.
 // It verifies the peer certificate using the root certificates associated with its trust domain.
 func (v *PeerCertVerifier) VerifyPeerCert(rawCerts [][]byte, _ [][]*x509.Certificate) error {
+	spiffeLog.Debugf("Verifying %d peer certificates", len(rawCerts))
 	if len(rawCerts) == 0 {
 		// Peer doesn't present a certificate. Just skip. Other authn methods may be used.
+		spiffeLog.Infof("Peer didn't present certificate")
```

Comment:
This log is going to be highly confusing in a real deployment. Which peer? Where did it not present cert? Is it intended (because it's using jwt) or an error?

I think we are logging at the wrong level most likely. I also suspect this will log on every CSR request, as those will of course not have certs?

Reply:
@howardjohn I was also concerned about logging and then returning `nil`.  If I remove that log statement, is the rest OK?

Without this PR, a request with no cert logs
```
error	ads	Failed to authenticate client &{[::1]:57854 {{772 true false 4865 h2 true localhost [] [] [] [] 0x1410950 []} {PrivacyAndIntegrity}}}
```

With this line, a request with no cert logs
```
info	spiffe	Peer didn't present certificate
error	ads	Failed to authenticate client &{[::1]:57854 {{772 true false 4865 h2 true localhost [] [] [] [] 0x1410950 []} {PrivacyAndIntegrity}}}
```

The current message is a bit opaque to me, but I couldn't figure out where to improve it.  I am OK with leaving it alone.

Reply:
In my opinion the spiffe function is not responsible for logging. Instead it can return an error why it didn't not authenticate and the caller can chose to log it or not.

Having two separate log lines for the same error is had practice in my opinion (they will not be sequential in a real system with parallelism) and the caller may not actually want to log the error if they authenticate with a token on cert auth failing

Reply:
@howardjohn I have moved the code out of the spiffe function.  Previously *nothing* from Istio was in the call stack on spiffe certificate verification failure.  I wrapped the spiffe function, does this version appeal to you?

I feel *something* should log.  Troubleshooting TLS connection failure without logs is painful.  I spent half an hour with Wireshark and couldn't figure it out.